internal/capacity/retrieve.go

```diff
@@ -21,55 +21,115 @@
 package capacity
 
 import (
+	"github.com/packethost/packngo"
 	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
 )
 
 func (c *Client) Retrieve() *cobra.Command {
 	var (
-		checkFacility bool
-		checkMetro    bool
+		metros, facilities, plans, locs []string
+		metro, facility, plan           string
 	)
 	// retrieveCapacitiesCmd represents the retrieveCapacity command
 	var retrieveCapacityCmd = &cobra.Command{
-		Use:     "get",
+		Use:     `get {-m [metros,...] | -f [facilities,...]} -P [plans,...]`,
 		Aliases: []string{"list"},
-		Short:   "Returns a list of facilities or metros and plans with their current capacity.",
-		Long: `Example:
-Retrieve capacities:
-metal capacity get { --metro | --facility }
-`,
+		Short:   "Returns a list of facilities or metros and plans with their current capacity, optionally filtered by given locations and plans.",
+		Example: `metal capacity get -m sv,ny,da -P c3.large.arm,c3.medium.x86`,
+
 		RunE: func(cmd *cobra.Command, args []string) error {
-			cmd.SilenceUsage = true
 			var err error
-			lister := c.Service.List
-			fieldName := "Facility"
+			var lister func() (*packngo.CapacityReport, *packngo.Response, error)
+			var locationField string
+
+			if metro != "" {
+				metros = append(metros, metro)
+			}
+			if facility != "" {
+				facilities = append(facilities, facility)
+			}
+
+			if (len(facilities) > 0) == (len(metros) > 0) {
+				return errors.New("Either facilities or metros should be set")
+			}
+			cmd.SilenceUsage = true
 
-			if checkMetro {
-				fieldName = "Metro"
+			if plan != "" {
+				plans = append(plans, plan)
+			}
+
+			if len(facilities) > 0 {
+				lister = c.Service.List
+				locationField = "Facility"
+				locs = append(locs, facilities...)
+			} else if len(metros) > 0 {
 				lister = c.Service.ListMetros
+				locationField = "Metro"
+				locs = append(locs, metros...)
 			}
 
 			capacities, _, err := lister()
 			if err != nil {
 				return errors.Wrap(err, "Could not get Capacity")
 			}
 
-			header := []string{fieldName, "Plan", "Level"}
-			requiredDataFormat := [][]string{}
+			header := []string{locationField, "Plan", "Level"}
+			data := [][]string{}
 
 			for locCode, capacity := range *capacities {
 				for plan, bm := range capacity {
-					loc := []string{}
-					loc = append(loc, locCode, plan, bm.Level)
-					requiredDataFormat = append(requiredDataFormat, loc)
+					if len(locs) > 0 {
```

Comment:
We can move the `loc` filtering to the previous loop, we can `continue` from the `locCode` loop if `locCode` is not in the `locs` list.

It may help to create a `contains` helper function rather than to create the comparison loop twice.